### DISCUSSION TITLE
how to handle a figure

### DISCUSSION THREAD

**Discussion by ryo3ryosuu** [BCG-Gamma/facet#207] [Q&A] [answered]
Could you answer the following questions.

Q1. how can I save a figure such as a matrix and dendrogram?
Q2. how can I change the size of the figure?

**Comment by jason-bentley** [comment, marked as answer]
Hi @ryo3ryosuu welcome to FACET discussions and thanks for your questions!

As the matrixplots and dendrograms are plotted using matplotlib, figure modifications can be performed as they would for any matplotlib plot. In the case of figure sizes there is a long standing thread on stackexchange [here](https://stackoverflow.com/questions/332289/how-do-you-change-the-size-of-figures-drawn-with-matplotlib) as a good starting place and for saving matplotlib figures you can see the documentation [here](https://matplotlib.org/3.1.1/api/_as_gen/matplotlib.pyplot.savefig.html). One solution applying this to FACET inspection plots is provided below.

Thanks!

***

After fitting your LearnerInspector() and for example, extracting the synergy matrix: 
```python
from pytools.viz.matrix import MatrixDrawer
synergy_matrix = inspector.feature_synergy_matrix()
```

you can set the figure size as shown before plotting in your notebook (assuming you have imported pyplot, i.e., ```import matplotlib.pyplot as plt```):
```python
plt.rcParams["figure.figsize"] = (14, 10)
MatrixDrawer(style="matplot%").draw(synergy_matrix, title="Synergy Matrix")
```

You can then save a copy of the plot as follows:
```python
plt.savefig("synergy_matrix.png", bbox_inches="tight", pad_inches=0)
```
